### DISCUSSION TITLE
Embedded search query with backlinks query logic

### DISCUSSION THREAD

**Discussion by zachmueller** [BryanJenksCommunity/FAQ#31] [Ideas & Suggestions]
I recently started using Obsidian and came across your YouTube channel which helped immensely in getting ramped up on using the app in more scalable ways. Thanks for the content you're publishing! Earlier today, I finally found a solution to a problem that had made an implementation of an idea less than perfect.

#### My use case

I'm using Obsidian for work and after I grasped your style of leveraging links vs tags, I realized I could finally implement an idea I've been wanting to solve for a long time. The idea has to do with keeping track of questions I have for specific people. The process needs to be simple to create the question, easy to differentiate between answered vs unanswered questions, but also scale to handling scenarios where one question could be answered by multiple people (and not wanting to manually duplicate the question/status in different places).

Leveraging the embedded query searches, I can create a new note, add the content of the question within the note, link it to all the relevant people I might ask that question to, then tag the note as being something I want to "check in" with a person on as well as tagging that the question is not yet answered. With templates, the initial tags get rolled in automatically, then I add in the question itself and links to the people. On each person's note (I have dedicated notes for each person I interact with frequently at work), there is a section at the bottom that looks for notes linked to it that are tagged both as being a "check in" note as well as are still in an "open" status. Anytime I meet with the person, I can call up their note and bring up any pending questions I've had for them.

#### The edge cases

There doesn't seem to be (using Obsidian version 0.10.11) a syntax for specifically searching for "notes that link to X note". I noticed on occasion the simplest/intuitive way of searching for backlinks (`[[Note name]]`) failed in certain situations. When searching for `[[Bryan Jenks]]`, Obsidian search treats each chunk (`[[Bryan` and `Jenks]]`) as a unique string to search for. Thus, if notes happen to include links to similar pages (with the start and end of their names matching plus the other parts somewhere in the note), it will include them in the search results.

#### Example test searches and Solution details

You can test an example of this with a couple test notes with the following content:

Note **`DO NOT FIND ME`** with content:
```
[[datetime]]

or a [[just a tag]]?
```

Note **`YES PLEASE FIND ME`** with content:
```
Say [[date tag|Hello World]]!
```

Search options:
1. `[[date tag]]` - erroneously includes "similar" links
2. `"[[date tag]]"` - erroneously excludes any aliased links
3. `("[[date tag]]" OR "[[date tag|")` - properly excludes "similar" links, the parenthases are only necessary if combining with other search terms

So, now my template for the Person page includes this embedded search:
```query
("[[{{title}}]]" OR "[[{{title}}|") tag:#people/checkin tag:#open
```

---

Hope someone finds this helpful!

Bryan, I'm curious whether you've run into a similar situation with searching for backlinks in Obsidian before? And am curious if you have any insight on whether the devs plan to implement specific syntax for searching for links (that properly excludes the similar results while properly includes the aliased results).

**Comment by tallguyjenks** [comment]
This is a great reasource! Thank you for your contribution! the solution for the alias part of the search is 👌🏻💯 

As for insight i have no special insight than anyone else in the Obsidian Server, but some basic additions will be coming to the queries sometime soon. I spoke to @lishid about sorting the query results ASC & DESC and he said:

![image](https://user-images.githubusercontent.com/29872822/107173798-8483f100-697d-11eb-9636-ea8c4ff684c7.png)

Thank you again for the great contribution!

**Reply by zachmueller** [reply]
Anytime! I'll post more ideas as I make more use of Obsidian.

And thanks for that! Sorting the query results will certainly be helpful for prioritizing which items show up first. Hopefully sometime I'll get around to exploring the discussions on the Obsidian Server more directly too.
